tests/Unit/IO/H5/Test_CombineH5.py

```diff
@@ -0,0 +1,159 @@
+                ],
+            )
+
+        # Store initial connectivity data
+        self.initial_h5_connectivity = self.vol_file1.get_tensor_component(
+            self.observation_ids[0], "connectivity"
+        ).data
+
+        self.h5_file1.close()
+
+        # Writing ElementVolume data and TensorComponent Data to second file
+        self.h5_file2 = spectre_h5.H5File(file_name=self.file_name2, mode="a")
+        self.tensor_component_data2 = np.array([[1.0], [0.13], [0.16]])
+        self.h5_file2.insert_vol("/element_data", version=0)
+        self.h5_file2.close_current_object()
+        self.vol_file2 = self.h5_file2.get_vol(path="/element_data")
+
+        self.element_vol_data_file_2 = [
+            ElementVolumeData(
+                element_name=grid_names2[0],
+                components=[
+                    TensorComponent(
+                        "field_1", DataVector(self.tensor_component_data2[i])
+                    ),
+                    TensorComponent(
+                        "field_2",
+                        DataVector(self.tensor_component_data2[i + 1]),
+                    ),
+                ],
+                extents=3 * [2],
+                basis=3 * [basis],
+                quadrature=3 * [quad],
+            )
+            for i, observation_id in enumerate(self.observation_ids)
+        ]
+
+        # Write extents and tensor volume data to the volfile
+        for i, observation_id in enumerate(self.observation_ids):
+            self.vol_file2.write_volume_data(
+                observation_id,
+                observation_values[observation_id],
+                [
+                    self.element_vol_data_file_2[i],
+                ],
+            )
+        self.h5_file2.close()
+
+    def tearDown(self):
+        self.h5_file1.close()
+        self.h5_file2.close()
+        if os.path.isfile(self.file_name1):
+            os.remove(self.file_name1)
+        if os.path.isfile(self.file_name2):
+            os.remove(self.file_name2)
+        if os.path.isfile(self.output_file + "0.h5"):
+            os.remove(self.output_file + "0.h5")
+
+    def test_combine_h5(self):
+        # Run the combine_h5 command and check if any feature (for eg.
+        # connectivity length has increased due to combining two files)
+
+        combine_h5(self.file_name1[:-4], "element_data", self.output_file)
+        h5_output = spectre_h5.H5File(
+            file_name=self.output_file + "0.h5", mode="a"
+        )
+        output_vol = h5_output.get_vol(path="/element_data")
+
+        # Extracts the connectivity data from the volume file
+        # If length of final connectivity is more, combine_h5
+        # has successfully merged the two HDF5 files.
+
+        final_h5_connectivity = output_vol.get_tensor_component(
+            self.observation_ids[0], "connectivity"
+        ).data
+
+        assert len(self.initial_h5_connectivity) < len(final_h5_connectivity)
```

Comment:
since you went through all this trouble to write the two files, you can now also read the combined file and check its content